addons/stock/models/stock_inventory.py

```diff
@@ -325,18 +318,14 @@ def _compute_difference(self):
 
     @api.depends('inventory_date', 'product_id.stock_move_ids', 'theoretical_qty', 'product_uom_id.rounding')
     def _compute_outdated(self):
-        grouped_quants = self.env['stock.quant'].read_group(
-            [('product_id', 'in', self.product_id.ids), ('location_id', 'in', self.location_id.ids)],
-            ['product_id', 'location_id', 'lot_id', 'package_id', 'owner_id', 'quantity:sum'],
-            ['product_id', 'location_id', 'lot_id', 'package_id', 'owner_id'],
-            lazy=False)
-        quants = {
-            (quant['product_id'][0],
-            quant['location_id'][0],
-            quant['lot_id'] and quant['lot_id'][0],
-            quant['package_id'] and quant['package_id'][0],
-            quant['owner_id'] and quant['owner_id'][0]): quant['quantity']
-            for quant in grouped_quants
+        quants_groups = self.inventory_id._get_quantities()
+        quants = {(
+            quants_group['product_id'] and quants_group['product_id'][0] or False,
+            quants_group['location_id'] and quants_group['location_id'][0] or False,
+            quants_group['lot_id'] and quants_group['lot_id'][0] or False,
+            quants_group['package_id'] and quants_group['package_id'][0] or False,
+            quants_group['owner_id'] and quants_group['owner_id'][0] or False):
+            quants_group['quantity'] for quants_group in quants_groups
```

Comment:
maybe _get_quantities could return this struct ? (quants)

and in _get_inventory_lines_values you could do
``` python
        for (product_id, location_id, lot_id, package_id, owner_id), quantity in quants:
            line_values = {
                'inventory_id': self.id,
                'product_qty': 0 if self.prefill_counted_quantity == "zero" else quantity,
                'theoretical_qty': quantity,
                'prod_lot_id': lot_id.id,
                'partner_id': owner_id.id,
                'product_id': product_id.id,
                'location_id': location_id.id,
                'package_id': package_id.id,
            }
```